plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1398,6 +1398,22 @@ public Optional<ConnectorOutputMetadata> finishInsert(
             transactionLogWriter.flush();
             writeCommitted = true;
             writeCheckpointIfNeeded(session, new SchemaTableName(handle.getSchemaName(), handle.getTableName()), checkpointInterval, commitVersion);
+
+            if (isCollectExtendedStatisticsColumnStatisticsOnWrite(session) && !computedStatistics.isEmpty()) {
+                // TODO: Add synchronization when version conflict for INSERT is resolved.
+                Optional<ExtendedStatistics> extendedStatistics = statisticsAccess.readExtendedStatistics(session, handle.getLocation());
+                Optional<Set<String>> analyzedColumns = extendedStatistics.flatMap(ExtendedStatistics::getAnalyzedColumns);
+                Optional<Instant> maxFileModificationTime = dataFileInfos.stream()
```

Comment:
maxFileModificationTime probably doesn't need to be optional, since for empty insert there should be no stats collected.
@findinpath does empty insert still create a transaction log entry?

Reply:
I checked and insert creates transaction log entry for that case. 
I will add check for that case, but I would leave Optional here to meet updateTableStatistics arguments, as in general maxFileModificationTime can be empty (for example during ANALYZE when it can't be retrieved from computed statistics).

Reply:
https://github.com/trinodb/trino/pull/16125#issuecomment-1431451234